### PR TITLE
Replace fvdl with ffx, allow test without install

### PR DESCRIPTION
**Pull request by djkoloski**
Along with replacing fvdl uses with the equivalent ffx commands, this also switches from using the install path for libstd-*.so and libtest-*.so to using the build directory (now passed on the command line). The user no longer needs to run x.py install before running tests now, and the correct libstd and libtest are detected on run instead of startup so the test runner can handle recompilations after starting the testing environment.

r? @tmandry

### CODE REVIEW

File: src/ci/docker/scripts/fuchsia-test-runner.py [hunk 1]
```diff
@@ -502,6 +451,19 @@ def run(self, args):
...
+
+        if not libstd_paths:
+            raise Exception(f"Failed to locate libstd (in {self.rustlibs_dir()})")
+
+        if not libtest_paths:
+            raise Exception(f"Failed to locate libtest (in {self.rustlibs_dir()})")
+
+        libstd_name = os.path.basename(libstd_paths[0])
+        libtest_name = os.path.basename(libtest_paths[0])
+
```
Comment (tmandry) [inline src/ci/docker/scripts/fuchsia-test-runner.py:466]: What about pretending the libs path here so the template doesn't have to also know about it?

File: src/ci/docker/scripts/fuchsia-test-runner.py [hunk 2]
```diff
@@ -275,87 +256,63 @@ def start(self):
...
-            authfile.write(authfile_contents)
-
-        # Start ffx isolation
-        self.log_info("Starting ffx isolation...")
-        self.start_ffx_isolation()
-
-        # Start emulator (this will generate the vdl output)
-        self.log_info("Starting emulator...")
+        # FIXME: condition --accel hyper on target arch matching host arch
         subprocess.check_call(
```
Comment (tmandry) [inline src/ci/docker/scripts/fuchsia-test-runner.py:297]: Should this be removed?

File: src/ci/docker/scripts/fuchsia-test-runner.py [hunk 3]
```diff
@@ -275,87 +256,63 @@ def start(self):
...
             ],
+            env=ffx_env,
             stdout=self.subprocess_output(),
             stderr=self.subprocess_output(),
         )
-        authfile_contents = subprocess.check_output(
+
+        # Start emulator
+        self.log_info("Starting emulator...")
+        product_bundle = "terminal.qemu-" + self.triple_to_arch(self.target)
```
Comment (tmandry) [inline src/ci/docker/scripts/fuchsia-test-runner.py:284]: remove?

**Comment by tmandry** [comment]
looks like there's an unintentional submodule change

File: src/ci/docker/scripts/fuchsia-test-runner.py (hunk 2 above)
Comment (djkoloski) [reply src/ci/docker/scripts/fuchsia-test-runner.py:297]: I think we want to leave this in, `--accel hyper` is almost always what we want. It just might not always work. I haven't tested what happens if we `--accel hyper` when the architectures don't match.

File: src/ci/docker/scripts/fuchsia-test-runner.py (hunk 3 above)
Comment (djkoloski) [reply src/ci/docker/scripts/fuchsia-test-runner.py:284]: Not sure what you mean by this.

Comment (tmandry) [reply src/ci/docker/scripts/fuchsia-test-runner.py:284]: nevermind, this was supposed to be on an earlier diff.

**Pull request merged**